EIPS/eip-1014.md

```diff
@@ -10,14 +10,80 @@ created: 2018-04-20
 
 ### Specification
 
-Adds a new opcode at 0xf5, which takes 4 stack arguments: endowment, memory_start, memory_length, salt. Behaves identically to CREATE, except using `keccak256(msg.sender ++ salt ++ init_code)[12:]` instead of the usual sender-and-nonce-hash as the address where the contract is initialized at.
+Adds a new opcode at 0xf5, which takes 4 stack arguments: endowment, memory_start, memory_length, salt. Behaves identically to CREATE, except using `keccak256( 0xff ++ address ++ salt ++ keccak256(init_code)))[12:]` instead of the usual sender-and-nonce-hash as the address where the contract is initialized at. The `salt` is a 32-byte stack item.
+
+The coredev-call at 2018-08-10 decided to use the formula above. 
+
 
 ### Motivation
 
 Allows interactions to (actually or counterfactually in channels) be made with addresses that do not exist yet on-chain but can be relied on to only possibly eventually contain code that has been created by a particular piece of init code. Important for state-channel use cases that involve counterfactual interactions with contracts.
 
-#### Option 2
+### Rationale about address formula 
+
+* Ensures that addresses created with this scheme cannot collide with addresses created using the traditional `keccak256(rlp([sender, nonce]))` formula, as `0xff` can only be a starting byte for RLP for data many petabytes long.
+* Ensures that the hash preimage has a fixed size,
+
+This also has the side-effect of being able to possibly reuse the `keccak256(init_code)` from earlier calculation, either within a client or via `EXTCODEHASH` if the init-code is deployed on-chain. 
+
```

Comment:
So the thought was, that if it's a very large initcode, and the verification of a given contract needs to be cheap in gas, the `init_code` could be deployed on-chain. 

If a contract `A` wants to verify a given contract address `C`, instead of feeding it the initcode and then force `A` do calculate the `sha3(init_code)`, `A` could just use the `EXTCODEHASH(D)`. Then `A` would know that yes, the `initcode` matches what's in `D`.   Where `D` would be a piece of whitelisted `init_code`. 

But yeah, maybe I should remove that. It was a 'hmm maybe this is useful for some scenario somewhere' kind of thing.